Haneke/Cache.swift

```diff
@@ -33,7 +33,14 @@ extension HanekeGlobals {
     
 }
 
-public class Cache<T: DataConvertible where T.Result == T, T : DataRepresentable> {
+public class Cache<T: DataConvertible where T.Result == T, T : DataRepresentable>: HanekeCache<T, DiskCache, NSCache>{
+
+    public override init(name: String) {
+        super.init(name: name)
+    }
+}
+
+public class HanekeCache<T: DataConvertible, DiskCacheT, MemoryCacheT where T.Result == T, T : DataRepresentable, DiskCacheT: DiskCache, MemoryCacheT: NSCache> {
```

Comment:
Having `MemoryCacheT: NSCache` as generic constraint means that the memory cache **must** be a NSCache subclass, while subclassing NSCache might be not the main preference of a developer while building a custom memory caching system. I recommend to protocols DiskCache and MemoryCache to give complete implementation freedom to the developer.

Reply:
you are correct. but I need someone to ensure that this PR is going to be merged before I spend time on this.

Reply:
Seems fair. I'm not the maintainer so I cannot give you this guarantee, but the swift 4 branch was just merged.